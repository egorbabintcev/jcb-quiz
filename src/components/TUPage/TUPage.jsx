import React from 'react';
import styles from './TUPage.module.scss';
import prices from './prices.png';
import gloves from './gloves.png';
import chat from './chat.png';

const TUPage = () => (
  <div className={styles.wrapper}>
    <h3 className={styles.title}>Спасибо за прохождение теста!</h3>
    <p className={styles.subtitle}>
      Ответы переданы нашему специалисту, он скоро
      <br /> приступит к работе и свяжется с вами
    </p>
    <div className={styles.gifts}>
      <h4 className={styles.title}>Гарантированные подарки:</h4>
      <div className={styles.gifts_wrapper}>
        <div className={styles.gift}>
          <img src={prices} alt="" />
          <p className={styles.title}>
            <strong>3 варианта цены</strong>
            <br /> на оригинал, аналог или б/у
          </p>
        </div>
        <div className={styles.gift}>
          <img src={gloves} alt="" />
          <p className={styles.title}>
            <strong>Перчатки<br /> в подарок</strong>
          </p>
        </div>
        <div className={styles.gift}>
          <img src={chat} alt="" />
          <p className={styles.title}>
            <strong>Фото и видео запчастей</strong>
            <br /> пришлем в Whatsapp
          </p>
        </div>
      </div>
    </div>
  </div>
);

export default TUPage;
