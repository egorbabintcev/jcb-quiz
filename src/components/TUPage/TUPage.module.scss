@import 'assets/sass/utils/functions';

.title {
  font: {
    size: rem(32);
    weight: 300;
  }
  margin: 0;
  margin-bottom: rem(15);
}

.subtitle {
  color: #838B98;
  font: {
    size: rem(18);
    weight: 300;
  }
  line-height: 1.5;
  margin: 0;
  margin-bottom: rem(80);
}

.gifts {
  display: flex;
  flex-direction: column;

  .title {
    margin-bottom: rem(30);
    text-align: center;
  }

  &_wrapper {
    display: flex;
    justify-content: space-between;
  }
}

.gift {
  background: linear-gradient(270deg, rgba(254, 206, 106, 0.84) -26.67%, rgba(254, 155, 54, 0.84) 138.33%);
  border-radius: rem(8);
  max-height: rem(215);
  max-width: rem(210);
  min-height: rem(215);
  min-width: rem(210);
  position: relative;

  &::before {
    border: rem(1) solid #FEC26D;
    border-radius: rem(9);
    content: '';
    display: block;
    height: calc(100% + #{rem(10)});
    left: 50%;
    position: absolute;
    top: 50%;
    transform: translate(-50%, -50%);
    width: calc(100% + #{rem(10)});
  }

  .title {
    bottom: rem(25);
    font: {
      size: rem(14);
      weight: 300;
    }
    left: rem(9);
    margin-bottom: 0;
    position: absolute;
    right: rem(9);

    strong {
      font: {
        size: rem(16);
        weight: 500;
      }
    }
  }

  img {
    left: 50%;
    min-height: rem(110);
    position: absolute;
    transform: translateX(-50%);
    top: rem(20);
  }
}
