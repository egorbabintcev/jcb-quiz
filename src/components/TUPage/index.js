import TUPage from './TUPage';

export default TUPage;
