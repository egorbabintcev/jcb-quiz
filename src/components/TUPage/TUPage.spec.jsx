import React from 'react';
import { render } from '@testing-library/react';
import TUPage from './TUPage';

describe('ThankYouPage component', () => {
  it('Should render thanksgiving page with title and 3 gifts', () => {
    const { asFragment } = render(<TUPage />);
    expect(asFragment(<TUPage />)).toMatchSnapshot();
  });
})
