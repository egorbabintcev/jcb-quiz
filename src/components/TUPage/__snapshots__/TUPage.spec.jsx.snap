// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThankYouPage component Should render thanksgiving page with title and 3 gifts 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <h3
      class="title"
    >
      Спасибо за прохождение теста!
    </h3>
    <p
      class="subtitle"
    >
      Ответы переданы нашему специалисту, он скоро
      <br />
       приступит к работе и свяжется с вами
    </p>
    <div
      class="gifts"
    >
      <h4
        class="title"
      >
        Гарантированные подарки:
      </h4>
      <div
        class="gifts_wrapper"
      >
        <div
          class="gift"
        >
          <img
            alt=""
            src="prices.png"
          />
          <p
            class="title"
          >
            <strong>
              3 варианта цены
            </strong>
            <br />
             на оригинал, аналог или б/у
          </p>
        </div>
        <div
          class="gift"
        >
          <img
            alt=""
            src="gloves.png"
          />
          <p
            class="title"
          >
            <strong>
              Перчатки
              <br />
               в подарок
            </strong>
          </p>
        </div>
        <div
          class="gift"
        >
          <img
            alt=""
            src="chat.png"
          />
          <p
            class="title"
          >
            <strong>
              Фото и видео запчастей
            </strong>
            <br />
             пришлем в Whatsapp
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
